jib-core/src/main/java/com/google/cloud/tools/jib/frontend/BuildStepsRunner.java

```diff
@@ -44,62 +44,57 @@
    * @return a {@link BuildStepsRunner} for building to a registry
    * @throws CacheDirectoryCreationException if the {@code cacheDirectory} could not be created
    */
   public static BuildStepsRunner forBuildImage(
-      BuildConfiguration buildConfiguration,
-      SourceFilesConfiguration sourceFilesConfiguration,
-      Path cacheDirectory,
-      boolean useOnlyProjectCache)
+      BuildConfiguration buildConfiguration, SourceFilesConfiguration sourceFilesConfiguration)
       throws CacheDirectoryCreationException {
-    Initializer cacheInitializer = getCacheInitializer(cacheDirectory, useOnlyProjectCache);
     return new BuildStepsRunner(
         BuildSteps.forBuildToDockerRegistry(
-            buildConfiguration, sourceFilesConfiguration, cacheInitializer));
+            buildConfiguration, sourceFilesConfiguration, getCacheInitializer(buildConfiguration)));
   }
 
   /**
    * Creates a runner to build to the Docker daemon. Creates a directory for the cache, if needed.
    *
    * @param buildConfiguration the configuration parameters for the build
    * @param sourceFilesConfiguration the source/destination file configuration for the image
-   * @param cacheDirectory the directory to use for the cache
-   * @param useOnlyProjectCache if {@code true}, sets the base layers cache directory to be the same
-   *     as the application layers cache directory
    * @return a {@link BuildStepsRunner} for building to a Docker daemon
    * @throws CacheDirectoryCreationException if the {@code cacheDirectory} could not be created
    */
   public static BuildStepsRunner forBuildToDockerDaemon(
-      BuildConfiguration buildConfiguration,
-      SourceFilesConfiguration sourceFilesConfiguration,
-      Path cacheDirectory,
-      boolean useOnlyProjectCache)
+      BuildConfiguration buildConfiguration, SourceFilesConfiguration sourceFilesConfiguration)
       throws CacheDirectoryCreationException {
-    Initializer cacheInitializer = getCacheInitializer(cacheDirectory, useOnlyProjectCache);
     return new BuildStepsRunner(
         BuildSteps.forBuildToDockerDaemon(
-            buildConfiguration, sourceFilesConfiguration, cacheInitializer));
+            buildConfiguration, sourceFilesConfiguration, getCacheInitializer(buildConfiguration)));
   }
 
-  private static Initializer getCacheInitializer(Path cacheDirectory, boolean useOnlyProjectCache)
+  // TODO: Move this up to somewhere where defaults for cache location are provided and ownership is
+  // checked rather than in Caches.Initializer.
+  private static Initializer getCacheInitializer(BuildConfiguration buildConfiguration)
       throws CacheDirectoryCreationException {
-    if (!Files.exists(cacheDirectory)) {
-      try {
-        Files.createDirectory(cacheDirectory);
-
-      } catch (IOException ex) {
-        throw new CacheDirectoryCreationException(cacheDirectory, ex);
-      }
+    CacheConfiguration applicationLayersCacheConfiguration;
```

Comment:
Maybe use a ternary operator here so you can declare and initialize on the same line.